libs/blocks/gnav/gnav.js

```diff
@@ -106,9 +106,16 @@ class Gnav {
 
   loadSearch = async () => {
     if (this.onSearchInput) return;
-    const { onSearchInput, getHelpxLink } = await import('./gnav-search.js');
-    this.onSearchInput = debounce(onSearchInput, SEARCH_DEBOUNCE_MS);
-    this.getHelpxLink = getHelpxLink;
+
+    const gnavSearch = await import('./gnav-search.js');
+    this.getHelpxLink = gnavSearch.getHelpxLink;
+
+    if (this.searchType === 'contextual') {
+      const { default: onSearchInput } = await import('./gnav-contextual-search.js');
+      this.onSearchInput = debounce(onSearchInput, SEARCH_DEBOUNCE_MS);
+    } else {
+      this.onSearchInput = debounce(gnavSearch.onSearchInput, SEARCH_DEBOUNCE_MS);
+    }
```

Comment:
Nit: Seems weird to undo the destructuring just to be able to use `onSearchInput` for contextual. I would just use a different name like `onContextualSearchInput`